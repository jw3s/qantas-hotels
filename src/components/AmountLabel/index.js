import React from 'react';
import styled from 'styled-components';

type Props = {
  children: string | number,
  symbol?: string,
};
const Wrapper = styled.span`
  display: flex;
  margin: 7px;
`;
const Currency = styled.sup`
  font-size: 1.2rem;
`;
const Label = styled.label`
  font-size: 3rem;
`;
const AmountLabel = ({ children, symbol }: Props) => (
  <Wrapper>
    <Label>
      <Currency>{symbol}</Currency>
      {children}
    </Label>
  </Wrapper>
);
AmountLabel.defaultProps = {
  symbol: '$',
};

export default AmountLabel;
