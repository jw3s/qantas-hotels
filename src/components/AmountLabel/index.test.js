import React from 'react';
import renderer from 'react-test-renderer';

import AmountLabel from '.';

import 'jest-styled-components';

describe('<AmountLabel />', () => {
  it('<AmountLabel /> should render tree', () => {
    const tree = renderer.create(<AmountLabel>500</AmountLabel>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
