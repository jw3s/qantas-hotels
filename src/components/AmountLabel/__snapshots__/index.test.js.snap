// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmountLabel /> <AmountLabel /> should render tree 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 7px;
}

.c2 {
  font-size: 1.2rem;
}

.c1 {
  font-size: 3rem;
}

<span
  className="c0"
>
  <label
    className="c1"
  >
    <sup
      className="c2"
    >
      $
    </sup>
    500
  </label>
</span>
`;
